src/decays/decay_functions.cpp

```diff
@@ -0,0 +1,339 @@
+
+   const std::complex<double> F0AC1A =
+      - z*(1.+Sqr(z))/(Cube(1.-z)*(1.+z))*(72.*li4p + 96.*li4m - 128./3.*(li3p + li3m)*ln
+            + 28./3.*li2p*Sqr(ln) + 16./3.*li2m*Sqr(ln) + 1./18.*Power4(ln)
+            + 8./3.*zeta2*Sqr(ln) + 32./3.*zeta3*ln + 12.*zeta4)
+      + z/Sqr(1.-z)*(-56./.3*li3p - 64./3.*li3m + 16.*li2p*ln
+            + 32./3*li2m*ln + 20./3.*std::log1p(-z)*Sqr(ln) - 8./3.*zeta2*ln + 8./3.*zeta3)
+      + 2.*z*(1.+z)/(3.*Cube(1.-z))*Cube(ln);
+   const std::complex<double> F0AC2A = 2./tau*(f(tau) - tau*fprime(tau));
+
+   return (F0AC1A + F0AC2A*std::log(Sqr(mu/mq)))/F0A;
+}
+
+/* 2-loop QCD corrections to H->gamma gamma amplitude through scalar color triplet loop
+ * from hep-ph/0611266 */
+
+/* this functions could be improved to cover all range in r by linking
+ * to CHAPLIN library (https://arxiv.org/abs/1106.5739) */
+std::complex<double> delta_hAA_2loopQCD_for_squark_loop(double mH, double msq, double mu) noexcept {
+   const std::complex<double> r = Sqr(mH/msq);
+   if (std::abs(r) < 0.7) {
+      // eq. A.3
+      const std::complex<double> F01l = -1./3. - 2./45*r - 1./140.*Sqr(r) - 2./1575*Cube(r) - 1./4158*Power4(r);
+      const std::complex<double> F02la = -3./4. - 29./216.*r - 4973./226800.*Sqr(r) - 3137./882000.*Cube(r) - 1180367./2095632000.*Power4(r);
+      const std::complex<double> F02lb = -1./4. - 1./15.*r - 9./560.*Sqr(r) - 2./525.*Cube(r) - 5./5544.*Power4(r);
+      const std::complex<double> F02lc = - 3/.4*F01l;
+      return (F02la + (F02lb + F02lc)*std::log(Sqr(msq/mu)))/F01l;
+   }
+   else if (std::abs(r) > 1.5) {
+      // eq. A.6
+      const std::complex<double> F01l = 4./r + Sqr(2.*std::log(-r)/r);
+      const std::complex<double> F02la =
+         (14.-3*std::log(-r))/r
+         + (6. - 72./5*Sqr(zeta2) - 24.*zeta3 - 8.*(1.-zeta2-4.*zeta3)*std::log(-r)
+               +(17.-8*zeta2)*Sqr(std::log(-r)) - 5./3.*Cube(std::log(-r)) - 1./6.*Power4(std::log(-r))
+            )/Sqr(r);
+      const std::complex<double> F02lb =
+         (6.*std::log(-r) - 3.*Sqr(std::log(-r)))/Sqr(r);
+      const std::complex<double> F02lc = - 3/.4*F01l;
+      return (F02la + (F02lb + F02lc)*std::log(Sqr(msq/mu)))/F01l;
+   }
+   else {
+      return 0.;
+   }
+}
+
+std::complex<double> delta_AhAA_2loopQCD_for_squark_loop(double mAH, double msq, double mu) noexcept {
+   const double r = Sqr(mAH/msq);
+   if (r < 0.7) {
+      return 0.;
+   }
+   else if (r > 1.5) {
+      return 0.;
+   }
+   else {
+      return 0.;
+   }
+}
+
+// eq. 6 of https://arxiv.org/pdf/1109.5304.pdf
+std::complex<double> hgg_SM_loop_function(double x) noexcept {
+   if(is_zero(1.-x)) {
+      return Sqr(Pi/2.);
+   }
+   else if (x > 1) {
+      return Sqr(std::asin(1./std::sqrt(x)));
+   }
+   else {
+      return -0.25*Sqr(std::log((1.+std::sqrt(1.-x))/(1.-std::sqrt(1.-x))) - 1i*Pi);
+   }
+}
+
+unsigned int number_of_active_flavours(softsusy::QedQcd const& qedqcd, double m) noexcept
```

Comment:
Sorry, I have a pedantic remark according to this function: For sane models, this function is totally fine. However, if the quark masses are unconventional, the function may break. 

Anyway, it is easy to fix. One could write:
~~~.cpp
unsigned nf = 0;
if (m < qedqcd.displayMass(softsusy::mUp)) { nf++; }
if (m < qedqcd.displayMass(softsusy::mDown)) { nf++; }
...
return nf;
~~~

Reply:
You mean `>`, right?

Reply:
Done in 79f7d31f367018c830835afe9a1bbbfbafb67ccf.

Reply:
Many thanks! Of course. :)